failing `server=local` tests on Travis against master

From https://travis-ci.org/mozilla/fxa-js-client/jobs/126195384

> FAIL: #with redirectTo (14ms)
AssertionError: expected { Object (code, errno, ...) } to be falsy
    at assert.notOk (/home/travis/build/mozilla/fxa-js-client/node_modules/intern-geezer/assert.js:529:10)
    at onSettled (/home/travis/build/mozilla/fxa-js-client/components/p/p.js:357:9)
    at Object.flush [as _onImmediate] (/home/travis/build/mozilla/fxa-js-client/components/p/p.js:64:6)
    at processImmediate [as _immediateCallback] (timers.js:363:15)
fxa-auth-server.INFO: request.summary {"op":"request.summary","code":400,"errno":107,"rid":"1461785098098:testing-worker-linux-docker-fffc1007-3362-linux-12:3187:inj93pje:10023","path":"/v1/account/login","agent":"Mozilla/5.0 (Linux x64) node.js/0.10.44 v8/3.14.5.9","remoteAddressChain":["127.0.0.1"],"t":2,"uid":"00","service":"sync","redirectTo":"http://sync.firefox.com/after_reset","keys":"true","email":"test1461785098091@restmail.net"}
FAIL: #with resume (11ms)
AssertionError: expected { Object (code, errno, ...) } to be falsy
    at assert.notOk (/home/travis/build/mozilla/fxa-js-client/node_modules/intern-geezer/assert.js:529:10)
    at onSettled (/home/travis/build/mozilla/fxa-js-client/components/p/p.js:357:9)
    at Object.flush [as _onImmediate] (/home/travis/build/mozilla/fxa-js-client/components/p/p.js:64:6)
    at processImmediate [as _immediateCallback] (timers.js:363:15)

I believe this is due to sending the resume token and redirectTo to an auth server that does not yet support those fields. Those fields are added in https://github.com/mozilla/fxa-auth-server/pull/1232


Ref https://github.com/mozilla/fxa-js-client/pull/197

--- a/client/FxAccountClient.js
+++ b/client/FxAccountClient.js
@@ -175,12 +175,6 @@ define([
    *   @param {String} [options.reason]
    *   Reason for sign in. Can be one of: `signin`, `password_check`,
    *   `password_change`, `password_reset`, `account_unlock`.
-   *   @param {String} [options.redirectTo]
-   *   a URL that the client should be redirected to after handling the request
-   *   @param {String} [options.resume]
-   *   Opaque url-encoded string that will be included in the verification link
-   *   as a querystring parameter, useful for continuing an OAuth flow for
-   *   example.
    *   @param {Object} [options.metricsContext={}] Metrics context metadata
    *     @param {String} options.metricsContext.flowId identifier for the current event flow
    *     @param {Number} options.metricsContext.flowBeginTime flow.begin event time
@@ -216,24 +210,16 @@ define([
             authPW: sjcl.codec.hex.fromBits(result.authPW)
           };
 
-          if (options.metricsContext) {
-            data.metricsContext = metricsContext.marshall(options.metricsContext);
+          if (options.service) {
+            data.service = options.service;
           }
 
           if (options.reason) {
             data.reason = options.reason;
           }
 
-          if (options.redirectTo) {
-            data.redirectTo = options.redirectTo;
-          }
-
-          if (options.resume) {
-            data.resume = options.resume;
-          }
-
-          if (options.service) {
-            data.service = options.service;
+          if (options.metricsContext) {
+            data.metricsContext = metricsContext.marshall(options.metricsContext);
           }
 
           return self.request.send(endpoint, 'POST', null, data)

--- a/client/FxAccountClient.js
+++ b/client/FxAccountClient.js
@@ -76,12 +76,6 @@ define([
    *   example.
    *   @param {String} [options.lang]
    *   set the language for the 'Accept-Language' header
-   *   @param {Object} [options.device={}] Device registration information
-   *     @param {String} options.device.name Name of device
-   *     @param {String} options.device.type Type of device (mobile|desktop)
-   *     @param {string} [options.device.callback] Device's push endpoint
-   *     @param {string} [options.device.publicKey] Public key used to encrypt push messages
-   *     @param {string} [options.device.authKey] Authentication secret used to encrypt push messages
    *   @param {Object} [options.metricsContext={}] Metrics context metadata
    *     @param {String} options.metricsContext.flowId identifier for the current event flow
    *     @param {Number} options.metricsContext.flowBeginTime flow.begin event time
@@ -114,22 +108,6 @@ define([
           var requestOpts = {};
 
           if (options) {
-            if (options.device) {
-              data.device = {
-                name: options.device.name,
-                type: options.device.type
-              };
-
-              if (options.device.callback) {
-                data.device.pushCallback = options.device.callback;
-              }
-
-              if (options.device.publicKey && options.device.authKey) {
-                data.device.pushPublicKey = options.device.publicKey;
-                data.device.pushAuthKey = options.device.authKey;
-              }
-            }
-
             if (options.service) {
               data.service = options.service;
             }
@@ -203,13 +181,6 @@ define([
    *   Opaque url-encoded string that will be included in the verification link
    *   as a querystring parameter, useful for continuing an OAuth flow for
    *   example.
-   *   @param {Object} [options.device={}] Device registration information
-   *     @param {String} [options.device.id] User-unique identifier of device
-   *     @param {String} [options.device.name] Name of device
-   *     @param {String} [options.device.type] Type of device (mobile|desktop)
-   *     @param {string} [options.device.callback] Device's push endpoint
-   *     @param {string} [options.device.publicKey] Public key used to encrypt push messages
-   *     @param {string} [options.device.authKey] Authentication secret used to encrypt push messages
    *   @param {Object} [options.metricsContext={}] Metrics context metadata
    *     @param {String} options.metricsContext.flowId identifier for the current event flow
    *     @param {Number} options.metricsContext.flowBeginTime flow.begin event time
@@ -244,31 +215,6 @@ define([
             email: result.emailUTF8,
             authPW: sjcl.codec.hex.fromBits(result.authPW)
           };
-
-          if (options.device) {
-            data.device = {};
-
-            if (options.device.id) {
-              data.device.id = options.device.id;
-            }
-
-            if (options.device.name) {
-              data.device.name = options.device.name;
-            }
-
-            if (options.device.type) {
-              data.device.type = options.device.type;
-            }
-
-            if (options.device.callback) {
-              data.device.pushCallback = options.device.callback;
-            }
-
-            if (options.device.publicKey && options.device.authKey) {
-              data.device.pushPublicKey = options.device.publicKey;
-              data.device.pushAuthKey = options.device.authKey;
-            }
-          }
 
           if (options.metricsContext) {
             data.metricsContext = metricsContext.marshall(options.metricsContext);

--- a/tests/lib/signIn.js
+++ b/tests/lib/signIn.js
@@ -9,11 +9,6 @@ define([
   'tests/lib/push-constants'
 ], function (tdd, assert, Environment, PushTestConstants) {
 
-  var DEVICE_CALLBACK = PushTestConstants.DEVICE_CALLBACK;
-  var DEVICE_ID = PushTestConstants.DEVICE_ID;
-  var DEVICE_NAME = PushTestConstants.DEVICE_NAME;
-  var DEVICE_TYPE = PushTestConstants.DEVICE_TYPE;
-
   with (tdd) {
     suite('signIn', function () {
       var ErrorMocks;
@@ -196,56 +191,6 @@ define([
               assert.equal(res.errno, 103);
             }
           );
-      });
-
-      test('#with new device', function () {
-        var email = 'test' + new Date().getTime() + '@restmail.net';
-        var password = 'iliketurtles';
-
-        return respond(client.signUp(email, password), RequestMocks.signUp)
-          .then(function () {
-            return respond(client.signIn(email, password, {
-              device: {
-                name: DEVICE_NAME,
-                type: DEVICE_TYPE,
-                callback: DEVICE_CALLBACK
-              },
-              reason: 'signin'
-            }), RequestMocks.signInNewDevice);
-          })
-          .then(
-            function (resp) {
-              var device = resp.device;
-              assert.ok(device.id);
-              assert.equal(device.name, DEVICE_NAME);
-              assert.equal(device.type, DEVICE_TYPE);
-              assert.equal(device.pushCallback, DEVICE_CALLBACK);
-            },
-            function (err) {
-              console.log(err);
-              assert.notOk();
-            }
-          );
-      });
-
-      test('#with existing device', function () {
-        var email = 'test' + new Date().getTime() + '@restmail.net';
-        var password = 'iliketurtles';
-
-        return respond(client.signUp(email, password), RequestMocks.signUp)
-          .then(function () {
-            return respond(client.signIn(email, password, {
-              device: {
-                id: DEVICE_ID,
-                name: DEVICE_NAME
-              },
-              reason: 'signin'
-            }), RequestMocks.signIn);
-          })
-          .then(function (resp) {
-            assert.ok(resp.uid);
-            assert.isUndefined(resp.device);
-          });
       });
 
       test('#with metricsContext metadata', function () {

--- a/tests/lib/signIn.js
+++ b/tests/lib/signIn.js
@@ -11,21 +11,19 @@ define([
 
   with (tdd) {
     suite('signIn', function () {
-      var ErrorMocks;
-      var RequestMocks;
       var accountHelper;
-      var client;
-      var mail;
       var respond;
+      var client;
+      var RequestMocks;
+      var ErrorMocks;
 
       beforeEach(function () {
         var env = new Environment();
-        ErrorMocks = env.ErrorMocks;
-        RequestMocks = env.RequestMocks;
         accountHelper = env.accountHelper;
-        client = env.client;
-        mail = env.mail;
         respond = env.respond;
+        client = env.client;
+        RequestMocks = env.RequestMocks;
+        ErrorMocks = env.ErrorMocks;
       });
 
       test('#basic', function () {
@@ -82,64 +80,6 @@ define([
             return respond(client.signIn(email, password, {reason: 'password_change'}), RequestMocks.signIn);
           });
       });
-
-      test('#with redirectTo', function () {
-        var user = 'test' + new Date().getTime();
-        var email = user + '@restmail.net';
-        var password = 'iliketurtles';
-        var opts = {
-          keys: true,
-          redirectTo: 'http://sync.firefox.com/after_reset',
-          service: 'sync'
-        };
-
-        return respond(client.signIn(email, password, opts), RequestMocks.signIn)
-          .then(function (res) {
-            assert.ok(res.uid);
-            return respond(mail.wait(user), RequestMocks.mailServiceAndRedirect);
-          })
-          .then(
-            function (emails) {
-              var code = emails[0].html.match(/code=([A-Za-z0-9]+)/)[1];
-              var redirectTo = emails[0].html.match(/redirectTo=([A-Za-z0-9]+)/)[1];
-
-              assert.ok(code, 'code is returned');
-              assert.ok(redirectTo, 'redirectTo is returned');
-
-            },
-            assert.notOk
-          );
-      });
-
-      test('#with resume', function () {
-        var user = 'test' + new Date().getTime();
-        var email = user + '@restmail.net';
-        var password = 'iliketurtles';
-        var opts = {
-          keys: true,
-          redirectTo: 'http://sync.firefox.com/after_reset',
-          resume: 'resumejwt',
-          service: 'sync'
-        };
-
-        return respond(client.signIn(email, password, opts), RequestMocks.signIn)
-          .then(function (res) {
-            assert.ok(res.uid);
-            return respond(mail.wait(user), RequestMocks.mailServiceAndRedirect);
-          })
-          .then(
-            function (emails) {
-              var code = emails[0].html.match(/code=([A-Za-z0-9]+)/)[1];
-              var resume = emails[0].html.match(/resume=([A-Za-z0-9]+)/)[1];
-
-              assert.ok(code, 'code is returned');
-              assert.ok(resume, 'resume is returned');
-
-            },
-            assert.notOk
-          );
-      });
-
 
       test('#incorrect email case', function () {
 

--- a/tests/lib/signUp.js
+++ b/tests/lib/signUp.js
@@ -9,10 +9,6 @@ define([
   'tests/lib/push-constants'
 ], function (tdd, assert, Environment, PushTestConstants) {
 
-  var DEVICE_CALLBACK = PushTestConstants.DEVICE_CALLBACK;
-  var DEVICE_NAME = PushTestConstants.DEVICE_NAME;
-  var DEVICE_TYPE = PushTestConstants.DEVICE_TYPE;
-
   with (tdd) {
     suite('signUp', function () {
       var accountHelper;
@@ -231,26 +227,6 @@ define([
             assert.equal(err.errno, 101);
           }
         );
-      });
-
-      test('#with new device', function () {
-        var email = 'test' + new Date().getTime() + '@restmail.net';
-        var password = 'iliketurtles';
-
-        return respond(client.signUp(email, password, {
-          device: {
-            name: DEVICE_NAME,
-            type: DEVICE_TYPE,
-            callback: DEVICE_CALLBACK
-          }
-        }), RequestMocks.signUpNewDevice)
-        .then(function (resp) {
-          var device = resp.device;
-          assert.ok(device.id);
-          assert.equal(device.name, DEVICE_NAME);
-          assert.equal(device.type, DEVICE_TYPE);
-          assert.equal(device.pushCallback, DEVICE_CALLBACK);
-        });
       });
 
       test('#with metricsContext metadata', function () {

--- a/tests/mocks/request.js
+++ b/tests/mocks/request.js
@@ -44,66 +44,15 @@ define([
       headers: {},
       body: '{ "uid": "0577e7a5fbf448e3bc60dacbff5dcd5c", "sessionToken": "27cd4f4a4aa03d7d186a2ec81cbf19d5c8a604713362df9ee15c4f4a4aa03d7d","keyFetchToken": "b1f4182d7e072567a1dbe682043a16932a84b7f4ca3b95e471a34806c87e4130"  }'
     },
-    signUpNewDevice: {
-      status: 200,
-      headers: {},
-      body: JSON.stringify({
-        device:{
-          id: DEVICE_ID,
-          name: DEVICE_NAME,
-          type: DEVICE_TYPE,
-          pushCallback: DEVICE_CALLBACK,
-          pushPublicKey: DEVICE_PUBLIC_KEY,
-          pushAuthKey: DEVICE_AUTH_KEY
-        },
-        sessionToken:'6544062365c5ebee16e3c5e15448139851583b5f5f7b6bd6d4a37bac41665e8a',
-        uid:'9c8e5cf6915949c1b063b88fa0c53d05',
-        verified:true,
-      })
-    },
     signIn: {
       status: 200,
       headers: {},
       body: '{"uid":"9c8e5cf6915949c1b063b88fa0c53d05","verified":true,"sessionToken":"6544062365c5ebee16e3c5e15448139851583b5f5f7b6bd6d4a37bac41665e8a"}'
     },
-    signInExistingDevice: {
-      status: 200,
-      headers: {},
-      body: JSON.stringify({
-        device:{
-          id: DEVICE_ID,
-          name: DEVICE_NAME,
-          type: DEVICE_TYPE,
-          pushCallback: DEVICE_CALLBACK,
-          pushPublicKey: DEVICE_PUBLIC_KEY,
-          pushAuthKey: DEVICE_AUTH_KEY
-        },
-        sessionToken:'6544062365c5ebee16e3c5e15448139851583b5f5f7b6bd6d4a37bac41665e8a',
-        uid:'9c8e5cf6915949c1b063b88fa0c53d05',
-        verified:true,
-      })
-    },
     signInFailurePassword: {
       status: 400,
       headers: {},
       body: '{"code":400,"message":"Incorrect password"}'
-    },
-    signInNewDevice: {
-      status: 200,
-      headers: {},
-      body: JSON.stringify({
-        device:{
-          id: DEVICE_ID,
-          name: DEVICE_NAME,
-          type: DEVICE_TYPE,
-          pushCallback: DEVICE_CALLBACK,
-          pushPublicKey: DEVICE_PUBLIC_KEY,
-          pushAuthKey: DEVICE_AUTH_KEY
-        },
-        sessionToken:'6544062365c5ebee16e3c5e15448139851583b5f5f7b6bd6d4a37bac41665e8a',
-        uid:'9c8e5cf6915949c1b063b88fa0c53d05',
-        verified:true,
-      })
     },
     signInWithKeys: {
       status: 200,

